frontend/src/app/modules/interventions/list/table/table.component.ts

```diff
@@ -1,44 +1,71 @@
 
-import { Intervention } from '@shared/domain/intervention.model';
+import { Intervention, SortDirection, InterventionsFilter } from '@shared/domain/intervention.model';
+import { InterventionsDatasource } from './interventions.datasource';
+import { InterventionsService } from '../../interventions.service';
+import { tap, startWith, switchMap, map } from 'rxjs/operators';
+import { combineLatest } from 'rxjs';
+import { ComponentWithSubscriptions } from '@shared/components/base';
+import { EkoRoutePaths } from '../../../../eko-route-paths';
 
 @Component({
-  selector: 'app-interventions-table',
+  selector: 'eko-interventions-table',
   templateUrl: './table.component.html',
   styleUrls: ['./table.component.scss'],
 })
-export class TableComponent implements OnChanges, AfterViewInit {
+export class TableComponent extends ComponentWithSubscriptions implements AfterViewInit, OnInit {
   displayedColumns = ['position', 'name', 'status', 'phone', 'date', 'description'];
-  dataSource: MatTableDataSource<Intervention>;
-
-  @Input() interventions: Intervention[] = [];
+  dataSource: InterventionsDatasource;
 
   @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
   @ViewChild(MatTable, { static: true }) table: MatTable<any>;
+  @ViewChild(MatSort, { static: true }) sort: MatSort;
 
-  constructor(private router: Router) {}
-
-  ngOnChanges(changes: any) {
-    this.checkInterventions(changes.interventions.currentValue, changes.interventions.previousValue);
+  constructor(private router: Router, private interventionsService: InterventionsService) {
+    super();
   }
 
-  private checkInterventions(currentInterventions?: Intervention[], previousInterventions?: Intervention[]) {
-    if (currentInterventions !== previousInterventions) {
-      this.dataSource = new MatTableDataSource<Intervention>(currentInterventions);
-      this.dataSource.paginator = this.paginator;
-    }
+  showDetails(intervention: Intervention) {
+    this.router.navigate([EkoRoutePaths.Interventions, intervention.id]);
   }
 
-  showDetails(intervention: Intervention) {
-    this.router.navigate(['interwencje', intervention.id]);
+  ngOnInit() {
+    this.dataSource = new InterventionsDatasource(this.interventionsService);
   }
 
   ngAfterViewInit() {
     // TODO: Workaround for https://github.com/angular/components/issues/8057
     // Remove when fixed. Debounces original ngOnDestroy way after route animation finishes.
     const tableOnDestroy = this.table.ngOnDestroy.bind(this.table);
     this.table.ngOnDestroy = () => setTimeout(tableOnDestroy, 1000);
+    this.subscriptions.add(this.subscribeToSortAndPageChanges());
+  }
+
+  private subscribeToSortAndPageChanges() {
+    const sort$ = this.sort.sortChange.pipe(
+      tap(() => (this.paginator.pageIndex = 0)),
+      startWith({ active: 'date', direction: 'desc' }),
+    );
+    const page$ = this.paginator.page.pipe(
+      startWith({
+        length: 100,
+        pageIndex: 1,
```

Comment:
Actually, we should start with `pageIndex: 0` I guess, as it will go through the pipe and get incremented?

Reply:
yeah, you're right